plugins/conditionalrestrictions.py

```diff
@@ -0,0 +1,210 @@
+#-*- coding: utf-8 -*-
+
+###########################################################################
+##                                                                       ##
+## Copyrights osmose project 2021                                        ##
+##                                                                       ##
+## This program is free software: you can redistribute it and/or modify  ##
+## it under the terms of the GNU General Public License as published by  ##
+## the Free Software Foundation, either version 3 of the License, or     ##
+## (at your option) any later version.                                   ##
+##                                                                       ##
+## This program is distributed in the hope that it will be useful,       ##
+## but WITHOUT ANY WARRANTY; without even the implied warranty of        ##
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         ##
+## GNU General Public License for more details.                          ##
+##                                                                       ##
+## You should have received a copy of the GNU General Public License     ##
+## along with this program.  If not, see <http://www.gnu.org/licenses/>. ##
+##                                                                       ##
+###########################################################################
+
+from modules.OsmoseTranslation import T_
+from plugins.Plugin import Plugin
+import re
+from datetime import date
+from modules.Stablehash import stablehash64
+
+class ConditionalRestrictions(Plugin):
+  def init(self, logger):
+    Plugin.init(self, logger)
+
+    self.ReYear = re.compile(r'20\d\d') # Update in 2099
+    self.currentYear = date.today().year
+
+    self.errors[33501] = self.def_class(item = 3350, level = 2, tags = ['highway', 'fix:chair'],
+        title = T_('Bad conditional restriction'),
+        detail = T_('''Conditional restrictions should follow `value @ condition; value2 @ condition2` syntax.
+Combined restrictions should follow `value @ (condition1 AND condition2)`.
+Parentheses `()` must be used around the condition if the condition itself contains semicolons `;`, i.e. `value @ (date;date)`.'''))
```

Comment:
Should there be a link to https://wiki.openstreetmap.org/wiki/Conditional_restrictions in the description?

Reply:
I'll add 
`resource="https://wiki.openstreetmap.org/wiki/Conditional_restrictions"`
to the two errors involving syntax when you're finished with the review, thanks!

Reply:
Done